### PR TITLE
[2.0.x Backport] Stream PostgresCol List() from row buffer (#7152)

### PR DESCRIPTION
**Pull request by acohen4**
* Stream PostgresCol List() from row buffer

* Set buffer size to 3 in collection tests

* Handling edge case where two rows at buffer boundary have the same sort value (non-primary key)

* Seeing what happens when reducing list buffer to size 10

* Fix watchers

* Adding 'NOTE' to comments

* Change Watch event interleaving to handoff strategy

* Remove 'greedy' option from list

* Limit batch queries by buffer size

* Bump default buffer size to 1000

### CODE REVIEW

**Review by PFedak — APPROVED** [review]

File: src/internal/collection/postgres_collection.go [hunk 1]
```diff
@@ -434,47 +516,84 @@ func (c *postgresReadOnlyCollection) Count() (int64, error) {
...
-		watcher.forwardNotifications(c.ctx, lastUpdated)
-	}()
+		return watcher.sendInitial(&watch.Event{
+			Key:      []byte(m.Key),
+			Value:    m.Proto,
+			Type:     watch.EventPut,
+			Template: c.template,
+			Rev:      m.UpdatedAt.Unix(),
+		})
+	}); err != nil && err != errutil.ErrBreak {
```
[the remark this answers is not in the document]
Comment (acohen4) [reply src/internal/collection/postgres_collection.go:565]: that's a good call. I'll make the change here and in master
